docs/topics/exceptions.rst

```diff
@@ -62,6 +62,17 @@ remain disabled. Those components include:
 
 The exception must be raised in the component's ``__init__`` method.
 
+InvalidOutput
+-------------
+
+.. exception:: InvalidOutput
+
+This exception can be raised by a downloader or spider middleware to
+indicate that some method returned a value not suported by the processing
+chain.
+See :ref:`topics-spider-middleware` and :ref:`topics-downloader-middleware`
+for a list of supported output values.
```

Comment:
I removed the exception from the docs and added an underscore to the name. I also modified the `DownloaderMiddlewareManager` class to raise `_InvalidOutput` too, not a good idea to raise a custom exception from the spider middleware and `AssertionError` from the downloader middleware IMHO.